src/phes_base.cpp

```diff
@@ -280,93 +298,148 @@ ExistingReservoir get_existing_reservoir(string name) {
+                        str(grid_square) + "_shapefile_tile.shp");
+  }
+  if (!use_tiled_bluefield && !use_tiled_rivers) {
+    string csv_filename =
+        file_storage_location + "input/existing_reservoirs/" + existing_reservoirs_csv;
+    if (!file_exists(csv_filename)) {
+      cout << "File " << csv_filename << " does not exist." << endl;
+      throw 1;
+    }
+    reservoirs = read_existing_reservoir_data(csv_filename.c_str());
 
-  filename = file_storage_location + "input/existing_reservoirs/" + existing_reservoirs_shp;
+    names = read_names(convert_string(file_storage_location + "input/existing_reservoirs/" +
+                                      existing_reservoirs_shp_names));
 
-  char *shp_filename = new char[filename.length() + 1];
-  strcpy(shp_filename, filename.c_str());
-  if (!file_exists(shp_filename)) {
-    search_config.logger.debug("No file: " + filename);
-    throw(1);
+    filenames.push_back(file_storage_location + "input/existing_reservoirs/" +
+                        existing_reservoirs_shp);
   }
-  SHPHandle SHP = SHPOpen(convert_string(filename), "rb");
-  if (SHP != NULL) {
-    int nEntities;
-    vector<vector<GeographicCoordinate>> relevant_polygons;
-    SHPGetInfo(SHP, &nEntities, NULL, NULL, NULL);
 
-    SHPObject *shape;
-    for(int i = 0; i<nEntities; i++){
-      shape = SHPReadObject(SHP, i);
-      if (shape == NULL) {
-        fprintf(stderr, "Unable to read shape %d, terminating object reading.\n", i);
-      }
-      int idx = -1;
-      for (uint r = 0; r < reservoirs.size(); r++) {
-        if (reservoirs[r].identifier == names[i]) {
-          idx = r;
+  for (string filename : filenames) {
+    if (!file_exists(filename)) {
+      search_config.logger.error("No file: " + filename);
+      throw(1);
+    }
+    SHPHandle SHP = SHPOpen(convert_string(filename), "rb");
+    DBFHandle DBF = DBFOpen(convert_string(filename), "rb");
+    bool tiled_bluefield = use_tiled_bluefield && SHP->nShapeType == SHPT_POLYGON;
+    bool tiled_river = use_tiled_rivers && SHP->nShapeType == SHPT_ARC;
+    bool csv_names = !tiled_bluefield && !tiled_river;
+    int dbf_field = 0, dbf_name_field = 0, dbf_elevation_field = 0;
+    if (tiled_river) {
+      dbf_field = DBFGetFieldIndex(DBF, string("DIS_AV_CMS").c_str());
+      dbf_name_field = DBFGetFieldIndex(DBF, string("River_name").c_str());
+    }
+    if (tiled_bluefield) {
+      dbf_field = DBFGetFieldIndex(DBF, string("Vol_total").c_str());
+      dbf_elevation_field = DBFGetFieldIndex(DBF, string("Elevation").c_str());
+      dbf_name_field = DBFGetFieldIndex(DBF, string("Lake_name").c_str());
+    }
+    if (SHP != NULL) {
+      int nEntities;
+      SHPGetInfo(SHP, &nEntities, NULL, NULL, NULL);
+
+      for (int i = 0; i < nEntities; i++) {
+        SHPObject *shape;
+        shape = SHPReadObject(SHP, i);
+        if (shape == NULL) {
+          fprintf(stderr, "Unable to read shape %d, terminating object reading.\n", i);
```

Comment:
Object reading doesn't seem to terminate if shape == NULL, error is just printed

Reply:
Fixed